app/models/grda_warehouse/system_cohorts/currently_homeless.rb

```diff
@@ -0,0 +1,125 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module GrdaWarehouse::SystemCohorts
+  class CurrentlyHomeless < Base
+    def cohort_name
+      'Currently Homeless'
+    end
+
+    def sync
+      add_missing_clients
+
+      remove_housed_clients
+      remove_inactive_clients
+      remove_no_longer_meets_criteria
+    end
+
+    private def add_missing_clients
+      # Newly identified (first homeless enrollment in the past 2 years)
+      # for each client with ongoing enrollments not on cohort
+      #   find max exit or move in prior to min start of ongoing
+      # if no exit or move in within the 2 years before min start of ongoing - Newly identified
+      # if exit was to permanent destination within 2 year range, or move in was within 2 year range  - Returned from housing
+      # else Returned from inactive unless no service in INACTIVE period
+
+      moved_in_ph = enrollment_source.ongoing.ph.
+        where(she_t[:move_in_date].lt(Date.current)).
+        select(:client_id)
+
+      candidate_enrollments = homeless_enrollment_source.
+        entry.
+        ongoing.
+        where.not(client_id: service_history_source.where(date: Date.yesterday, homeless: false).select(:client_id)).
+        where.not(client_id: moved_in_ph).
+        where.not(client_id: cohort_clients.select(:client_id)).
+        group(:client_id).minimum(:first_date_in_program)
+
+      previous_enrollments = homeless_enrollment_source.
+        exit.
+        where(client_id: candidate_enrollments.keys).
+        order(greatest(she_t[:last_date_in_program], she_t[:move_in_date])).
```

Comment:
homeless enrollments won't have a move-in-date

Reply:
Yeah... I think this should be any enrollment, not just homeless ones, and the move in date is irrelevant since if they are currently enrolled in ph, they will be caught above, and the exit from PH should always fall after the move in date, right?

Reply:
Exit can fall on or after move-in, and there may not be a move-in, even if they have an exit.... which would mean the enrollment wouldn't negate homelessness.

Reply:
I think this is correct now...
  If they have a move in date before the current date in an open PH enrollment, they are caught in `moved_in_ph`
  previous enrollments will have an `last_date_in_program`, so whether they are PH or not is irrelevant, and the time in the past will determine if this is a new or returning client.